### PR TITLE
Add a small note about Kubernetes Client configuration properties.

### PR DESCRIPTION
**Pull request by rafaeltuelho**
Small doc PR based on this thread https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/openshift-extension/near/234538945

### CODE REVIEW

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 1]
```diff
@@ -680,6 +680,18 @@ quarkus.kubernetes.readiness-probe.period=45s
...
+These underline client connection gets automatically configured according to your local connection (by `kubectl`).
+
+In certain situations, you may need to tweak these properties like when your cluster API endpoint uses a self-signed SSL Certificate. In this case, you need to explicitly configure the client to trust your cluster API certificate. You can achieve this by defining the following property:
+
+[source,properties]
+----
+quarkus.kubernetes-client.trust-certs=true
+----
+
+Note that the full list of Kubernetes client properties is available in the https://github.com/quarkusio/quarkus/blob/main/extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientBuildConfig.java[KubernetesClientBuildConfig] class.
```
Comment (machi1990) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:694]: We can use the generated documentation reference 
```suggestion
{generated-dir}/config/quarkus-kubernetes-client.adoc
```

Comment (machi1990) [reply docs/src/main/asciidoc/deploying-to-kubernetes.adoc:694]: See https://github.com/quarkusio/quarkus/blob/7a4201a057c3c52ac428db19b14594ee9824fe61/docs/src/main/asciidoc/datasource.adoc#reactive-datasource-configuration-reference for example on how to include it

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 2]
```diff
@@ -680,6 +680,20 @@ quarkus.kubernetes.readiness-probe.period=45s
 In this example `initial-delay` and `period` are fields of the type `Probe`.
 Below you will find tables describing all available types.
 
+==== Client Connection Configuration
+The underline Kubernetes client connection can be configured through properties defined by the Kubernetes Client runtime.
```
Comment (geoand) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:684]: Underlying?

**Review by geoand — COMMENTED** [review]
Thanks for this!

I added a small note

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/deploying-to-kubernetes.adoc:684]: I would say something like:

```
You may need to configure the connection to your Kubernetes cluster.
By default, it automatically uses the active _context_ used by `kubectl`. 
...
```

**Review by cescoffier — CHANGES_REQUESTED** [review]

**Comment by geoand** [comment]
When you are done, please squash the commits.

Thanks!

**Comment by rafaeltuelho** [comment]
> When you are done, please squash the commits.
> 
> Thanks!

Done!

**Review by geoand — APPROVED** [review]

**Review by cescoffier — APPROVED** [review]

**Pull request merged**